crates/ra_hir_expand/src/db.rs

```diff
@@ -181,6 +181,11 @@ pub(crate) fn parse_macro(
                         err,
                         node.value
                     );
+                    let mut parent = loc.kind.file_id().call_node(db);
+                    while let Some(node) = parent.clone() {
+                        log::warn!("parent: macro_call: {:#})", node.value);
+                        parent = node.file_id.call_node(db);
+                    }
```

Comment:
Hm, I am not sure what this while loop should be doing? Perhaps a comment would be helpful?

Reply:
Sure and fixed! It is used for printing out the parents of the failed expansion.